File: 151001/Timoshek/publisher/src/main/java/by/bsuir/egor/dto/CommentMapper.java
package by.bsuir.egor.dto;

import by.bsuir.egor.Entity.Comment;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

import java.util.List;

@Mapper
public interface CommentMapper {
    CommentMapper INSTANCE = Mappers.getMapper( CommentMapper.class );

    Comment commentResponseToToComment(CommentResponseTo responseTo);
    Comment commentRequestToToComment(CommentRequestTo requestTo);

    CommentRequestTo commentToCommentRequestTo(Comment comment);

    CommentResponseTo commentToCommentResponseTo(Comment comment);

    List<CommentResponseTo> listRequestToResponse(List<CommentRequestTo> requestList);
}

File: 151003/Klimovich/Discussion/Services/Mappers/CommentMapper.cs
using AutoMapper;
using Discussion.Models;
using Discussion.Models.DTO.RequestTo;
using Discussion.Models.DTO.ResponceTo;

namespace Discussion.Services.Mappers
{
    public class CommentMapper : Profile
    {
        public CommentMapper()
        {
            CreateMap<Comment, CommentRequestTo>().ReverseMap();
            CreateMap<Comment, CommentResponceTo>().ReverseMap();

        }
    }
}
